src/adaptor/merkleadaptor.rs

```diff
@@ -169,9 +170,11 @@ impl<const DEPTH: usize> HostOpSelector for MerkleChip<Fr, DEPTH> {
             .map(|x| ((Fr::from(x.value), Fr::from(x.op as u64)), Fr::zero()))
             .collect::<Vec<((Fr, Fr), Fr)>>();
 
-        assert!(total_used_instructions <= TOTAL_CONSTRUCTIONS);
+        assert!(k >= 22);
```

Comment:
if k >= 22 assert is by design, it is ok.
(Although this means if user try small K image with host cirecuit, need check it outside first, or it will panic in lib)